### PR TITLE
bpo-36232: Improve error message on dbm.open() when the db doesn't exist

### PR DESCRIPTION
**Pull request by rougeth**
https://bugs.python.org/issue36232

### CODE REVIEW

**Comment by rougeth** [comment]
The forced-push was just to rebuild on Azure. It was falling for reasons not related to the changes.

**Review by remilapeyre — APPROVED** [review]

File: Lib/dbm/__init__.py [hunk 1]
```diff
@@ -82,7 +82,8 @@ def open(file, flag='r', mode=0o666):
             # file doesn't exist and the new flag was used so use default type
             mod = _defaultmod
         else:
-            raise error[0]("need 'c' or 'n' flag to open new db")
+            raise error[0]("db file doesn't exist, "
```
Comment (brettcannon) [inline Lib/dbm/__init__.py]:
```suggestion
            raise error[0]("db file doesn't exist; "
```

**Review by brettcannon — CHANGES_REQUESTED** [review]

**Comment by brettcannon** [comment]
@rougeth mind adding a news file? (And make sure to thank yourself!) You can use blurb-it -- found in the Details link for the bedevere/news CI check -- to add one easily.

**Comment by rougeth** [comment]
I have made the requested changes; please review again.

**Comment by rougeth** [comment]
@brettcannon I was looking into the other news entries, found a typo and fixed it, is that ok?

**Review by brettcannon — APPROVED** [review]

File: Lib/dbm/__init__.py [hunk 2]
```diff
@@ -82,7 +82,8 @@ def open(file, flag='r', mode=0o666):
             # file doesn't exist and the new flag was used so use default type
             mod = _defaultmod
         else:
-            raise error[0]("need 'c' or 'n' flag to open new db")
+            raise error[0]("db file doesn't exist; "
+                           "use 'c' or 'n' flag to open new db")
```
Comment (brettcannon) [inline Lib/dbm/__init__.py]:
```suggestion
                           "use 'c' or 'n' flag to create a new db")
```

**Review by brettcannon — CHANGES_REQUESTED** [review]
One minor wording change.

**Comment by brettcannon** [comment]
I would have committed the change myself but GH says I don't have permission to.

**Comment by rougeth** [comment]
I have made the requested changes; please review again

**Comment by rougeth** [comment]
@brettcannon I may have blocked PR changes from other people (don't know/remember why :) I believe now it's ok.

**Comment by brettcannon** [comment]
Thanks!

**Pull request merged**